api/rosetta/data_balance_integration_test.go

```diff
@@ -180,19 +168,18 @@ func TestAPI_BalanceHandlesErrors(t *testing.T) {
 
 		request rosetta.BalanceRequest
 
-		wantStatusCode              int
-		wantRosettaError            meta.ErrorDefinition
-		wantRosettaErrorDescription string
-		wantRosettaErrorDetails     map[string]interface{}
+		checkError assert.ErrorAssertionFunc
 	}{
 		{
 			name:    "empty balance request",
 			request: rosetta.BalanceRequest{},
 
-			wantStatusCode:              http.StatusBadRequest,
-			wantRosettaError:            configuration.ErrorInvalidFormat,
-			wantRosettaErrorDescription: "blockchain identifier: blockchain field is empty",
-			wantRosettaErrorDetails:     nil,
+			checkError: checkRosettaError(
+				http.StatusBadRequest,
+				configuration.ErrorInvalidFormat,
+				"blockchain identifier: blockchain field is empty",
+				nil,
```

Comment:
I'm not sure if this will be controversial, but I would like you to completely remove all checks on the exact error message and details map. Why? Because those two fields are meant to be the human-readable counter-part to the error code and definition. Only the code of the error and the reference description should be programatically checked, because only they are programatically relevant.

Reply:
Oof! I'm hurting! 😢

I get your point. For instance, when I was covering the `malformed JSON` scenarios, I did not verify the error returned, because it would be, for instance, JSON unmarshaller internal error of the type `could not unmarshal value of type int to a field of type string` or `unexpected token: EOF, expected '}'` (for unclosed bracket; also totally guessing, I have no idea what the exact message was.
For that I only verified that the status code and the error.

For these I did, I think for two reasons:
1. It made it easier to discern where exactly the error happened (e.g. which validation path). The tests `name` field does also point to that, perhaps a little less precisely. Checking the wanted error description in the tests and the `rosetta` code would very quickly show us which scenario/code path did the test cover.
2. It spoke for the strictness/thoroughness of the tests. This is 90% just my psyche, but it is still true :)

A strong argument for keeping them would be to make **sure** we error on specific scenarios. For instance, we return status 400 + `ErrorInvalidFormat` for both of these requests:
- `index` and `hash` unset
- `index` set, but `hash` is of invalid length

Here the error description tells us precisely which condition we hit. Judging by the status and error alone, it's harder to see if the second test case was indeed covered.

However, I do think our test inputs are thorough and varied enough that we covered majority<sup>[1]</sup> of the inputs. So at this stage, I don't think there's much harm if we remove them.

Perhaps the major effect is that it would make navigation between tests and code a bit harder. At the moment you can see `"block identifier: hash field has the wrong length".. ah, here it is!`.

I still like them, even though I'm aware it's technically wrong 😄 


<sup>[1]</sup> \- I wanted to say all but you can never be sure it's all

Reply:
We don't need to check the description or the details to check if we error. As long as the correct _error definition_ is returned, it doesn't matter imo if the details and description distinguish between two cases. I would go so far to say that as long as the correct error definition is returned, the description and details don't matter at all; we errored correctly according to specs. So we must imo not check them.